### ISSUE TITLE
Thread safe add/delete etc

### ISSUE THREAD

**Issue by VladimirMarkelov** [VladimirMarkelov/clui#46]

**Comment by gernest** [comment]
@VladimirMarkelov can you please expand a little bit about this issue. I am thinking about using this library for a new personal project.

Implementing new features is the easiest way to get familiar with a project.

**Comment by VladimirMarkelov** [comment]
In the beginning I was going to allow to add/delete widgets on the fly. So, I added an issue to test how the library works in multi-threaded application. I did not get any severe trouble while I was testing the library and a few my small apps worked fine. But I kept the issue because I knew that the multi-threaded case is not thoroughly tested.

Anyway, I believe the problem still exists if someone tries to create/delete something inside a code called via 'go ....'.  By today, I ran into the issue once. Most of the widgets fire their events in the same thread, but. e.g, TableView uses 'go ...' to fire event (I guess, some of them can be called without 'go ..' and it fixes some troubles - it needs testing). It may result in weird screen effects when trying to create anything inside event listener. A quick example: create TableView, and add listener to its onAction event. Try to create simple confirmation dialog inside the function. At the time the new dialog displays the whole terminal windows may show a distorted picture. If you check simple FB2 reader - the link to it in README - to fix it, I used a workaround: I created confirmation dialog(to confirm deleting a book from library) at program start and inside listener I just do show and hide to manipulate. I looks ugly.

So, the first thing that is good to fix - make possible to create a new dialogs inside all(or at least inside as many as possible) internal event listeners.
